subscriber/check.go

```diff
@@ -0,0 +1,59 @@
+package subscriber
+
+import (
+	"context"
+
+	"github.com/docker/distribution/reference"
+	"github.com/grafeas/voucher"
+	"github.com/grafeas/voucher/cmd/config"
+	"github.com/grafeas/voucher/repository"
+)
+
+// check runs all checks for a given image.
+// Returns true if the required check(s) have passed and true if the check run needs to be retried.
+func (s *Subscriber) check(canonicalImageReference reference.Canonical) (bool, bool) {
+	var repositoryClient repository.Client
+	var err error
+
+	ctx, cancel := context.WithTimeout(context.Background(), s.cfg.TimeoutDuration())
+	defer cancel()
+
+	metadataClient, err := config.NewMetadataClient(ctx, s.secrets)
+	if nil != err {
+		s.log.Errorf("failed to create MetadataClient: %s", err)
+		return false, true
+	}
+	defer metadataClient.Close()
+
+	buildDetail, err := metadataClient.GetBuildDetail(ctx, canonicalImageReference)
+	if nil != err {
+		s.log.Warningf("could not get image metadata for %s: %s", canonicalImageReference, err)
+	} else {
+		if s.secrets != nil {
+			repositoryClient, err = config.NewRepositoryClient(ctx, s.secrets.RepositoryAuthentication, buildDetail.RepositoryURL)
+			if nil != err {
+				s.log.Warningf("failed to create repository client, continuing without git repo support: %s", err)
+			}
+		} else {
+			s.log.Warning("failed to create repository client, no secrets configured")
+		}
+	}
+
+	checksuite, err := config.NewCheckSuite(s.secrets, metadataClient, repositoryClient, s.cfg.RequiredChecks...)
+	if nil != err {
+		s.log.Errorf("failed to create CheckSuite: %s", err)
+		return false, true
+	}
+
+	var results []voucher.CheckResult
+
+	if s.cfg.DryRun {
```

Comment:
yeah so I opted not to bother doing any refactors here because I wasn't sure how this feature would pan out in the future! that being said i'm happy to generalize this in 1 spot if overall we're okay with this structure / layout and functionality

> also note that in server.go it uses if viper.GetBool("dryrun")

for this implementation I decided to pass that flag in earlier when setting up the config and am referencing this value in the struct directly. i'll standardize / make this the same as `server/check.go` one way or another!